new-client/src/plugins/Kir/KirLayerController.js

```diff
@@ -0,0 +1,304 @@
+      feature.set("kir_type", "feature");
+    });
+
+    this.model.layers.features.getSource().addFeatures(arr);
+    if (options.zoomToLayer) {
+      this.zoomToLayer(this.model.layers.features);
+    }
+
+    // Force rendering of buffer to next tick to prevent gui freeze.
+    // buffer is throttled for good reasons.
+    clearTimeout(this.renderDelay_tm);
+    this.renderDelay_tm = setTimeout(() => {
+      this.bufferFeatures(options);
+    }, 250);
+  };
+
+  removeFeature = (feature) => {
+    this.handleRemoveFeature(feature.ol_uid);
+    this.observer.publish("kir.search.remove", feature);
+  };
+
+  clearBeforeSearch = () => {
+    this.model.layers.features.getSource().clear();
+    this.model.layers.marker.setVisible(false);
+  };
+
+  handleFeatureClicks = (e) => {
+    if (this.model.windowIsVisible !== true) {
+      return;
+    }
+    let featureFound = null;
+    this.model.map.forEachFeatureAtPixel(e.pixel, (feature, layer) => {
+      if (!featureFound && layer === this.model.layers.features && feature) {
+        featureFound = feature;
+      }
+    });
+
+    if (featureFound) {
+      this.mark({ open: true, feature: featureFound });
+      this.observer.publish("kir.search.feature.selected", featureFound);
+    } else {
+      this.mark({ open: false, feature: featureFound });
+    }
+  };
+
+  handleRemoveFeature = (uid) => {
+    const feature = this.model.layers.features.getSource().getFeatureByUid(uid);
+    if (feature) {
+      this.model.layers.features.getSource().removeFeature(feature);
+    }
+  };
+
+  handleClearSearch = (data) => {
+    this.clearBeforeSearch();
+    this.model.layers.draw.getSource().clear();
+    this.model.layers.buffer.getSource().clear();
+  };
+
+  bufferFeatures = (options) => {
+    if (!this.bufferValue) {
+      this.getLayer("buffer").getSource().clear();
+      return;
+    }
+
+    const parser = new jsts.io.OL3Parser();
+    parser.inject(
+      Point,
+      LineString,
+      LinearRing,
+      Polygon,
+      MultiPoint,
+      MultiLineString,
+      MultiPolygon
+    );
+
+    let drawFeatures = this.getLayer("draw").getSource().getFeatures();
+
+    if (drawFeatures.length > 0) {
+      this.getLayer("buffer").getSource().clear();
+    }
+
+    let _bufferFeatures = [];
+
+    drawFeatures.forEach((feature) => {
+      let olGeom = feature.getGeometry();
+      if (olGeom instanceof Circle) {
+        olGeom = Polygon.fromCircle(olGeom, 0b10000000);
+      }
+      const jstsGeom = parser.read(olGeom);
+      const bufferedGeom = jstsGeom.buffer(this.bufferValue);
+
+      let bufferFeature = new Feature({
+        geometry: parser.write(bufferedGeom),
+      });
+      bufferFeature.set("owner_ol_uid", feature.ol_uid);
+      bufferFeature.set("kir_type", "buffer");
+
+      _bufferFeatures.push(bufferFeature);
+    });
+
+    const targetSource = this.getLayer("buffer").getSource();
+    targetSource.addFeatures(_bufferFeatures);
+  };
+
+  _getZoomOptions = () => {
```

Comment:
I like the separation between public and private methods. But rather then this custom "underscore" method, I'd prefer to use the correct [ECMAScript Private Class Fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields). (Basically, switch the `_` to the `#`.)

I comment only here but the comment is valid for all occurrences of `_foobar`.

Reply:
This is a matter of me being too old school, using old syntax. Now it's time to punch this "new" syntax into the spine.